test/test_decomp.jl

```diff
@@ -11,3 +11,22 @@ i = Index(2,"i")
 j = Index(2,"j")
 A = randomITensor(i,j)
 @test_throws ArgumentError factorize(A, i, dir="fakedir")
+
+
+@testset "Spectrum" begin
+  i = Index(100,"i")
+  j = Index(100,"j")
+
+  U,S,V = svd(rand(100,100))
+  S ./= norm(S)
+  A = ITensor(U*diagm(0=>S)*V', i,j)
+
+  _,_,_,_,_,spec = svd(A,i)
```

Comment:
This is a bit unfortunate. Maybe we want svd to return a struct, and then you can get the spectrum with `spec = svd(A,i).spec`? I think you can still use the syntax `U,S,V = svd(A,i)` if it returns a struct, for example Julia does this:

```julia
julia> A = randn(2,2)
2×2 Array{Float64,2}:
 2.45284  -0.276589
 2.73764   0.779181

julia> F = svd(A)
SVD{Float64,Float64,Array{Float64,2}}([-0.6508728185978933 -0.7591867846652984; -0.7591867846652984 0.6508728185978933], [3.6978397331941464, 0.72161329507932], [-0.9937883641647395 -0.11128650974296625; -0.11128650974296625 0.9937883641647395])

julia> U,S,V = F;

julia> U == F.U
true
```

Reply:
Yes I agree we want to keep the `U,S,V = svd(....)` syntax working if possible.

Does having the above behavior of unpacking a custom struct require iteration to be defined for that struct?

Reply:
I would guess that is how you define that behavior.

Reply:
This is how it's done for the SVD factorization in LinearAlgebra
```julia
# iteration for destructuring into components
Base.iterate(S::SVD) = (S.U, Val(:S))
Base.iterate(S::SVD, ::Val{:S}) = (S.S, Val(:V))
Base.iterate(S::SVD, ::Val{:V}) = (S.V, Val(:done))
Base.iterate(S::SVD, ::Val{:done}) = nothing
```
So I could implement similar factorization structs for `svd` and `eigenHermitian` which 
will also contain the truncation information (could be called `TruncSVD`,`TruncEigen` or something like this).

As a side note - would you agree with my comment above that `svd` should actually return only truncation error (in addition to U,S,V etc.) and not the whole `Spectrum` object?

Reply:
Also, I've noticed that while `svd(A::ITensor,...)` returns `U,S,V,u,v,spec`, `u` and `v` seem to never be used in the code base (it is always called as `U,S,V = svd(...)`). 
So maybe the order of return parameters (with or without a factorization struct) should be changed such that `U,S,V,spec = svd(...)`  works?

Actually is it necessary to return `u,v` from `svd` ? I find the logic cleaner if the task of finding the common indices is done by the caller (which seems to occur anyway in practice).

Reply:
We can return the indices last (or not return them, but I think there is not much downside in returning them). I just thought it was neat to return them since it is much harder to do it in C++.

I don't have too much of an opinion on what object should be output, but Spectrum might give some other useful information (like for BlockSparse tensors, maybe it can output the truncation per block?). Maybe Miles has more to say on that.